intake/source/csv.py

```diff
@@ -142,12 +142,23 @@ def _get_schema(self):
             return self._schema
 
         nrows = self._csv_kwargs.get("nrows")
-        self._csv_kwargs["nrows"] = 10
+        skipfooter = self._csv_kwargs.get("skipfooter")
```

Comment:
How about
```
nrows = self._csv_kwargs.pop("nrows")
skipfooter = self._csv_kwargs.pop("skipfooter")
df = self._get_partition(0)
if nrows:
    self._csv_kwargs["nrows"] = nrows
if skipfooter:
    self._csv_kwargs["skipfooter"] = skipfooter
```

Reply:
That wouldn't work in this case, on two counts:

1. `dict.pop(key)` fails and reaises a `KeyError` if `key` is not present in the dict. We don't know that `self._csv_kwargs` contains either `"nrows"` or `"skipfooter"` to begin with. To avoid a possible crash, we have to either use `dict.get` as in the PR code, or test first with, e.g., `if "skipfooter" in selv._csv_kwargs` first, which seems more awkward. Is there any particular reason you want to use `dict.pop` here?

2. If `"skipfooter"` is in `self._csv_kwargs`, I think we should add that number of rows to the 10 rows to be read, like I have done in line 148 of the PR code, and then remove the same number of rows from the bottom of the returned DataFrame, like I do in line 160. If `skipfooter` is set, it means that there are rows at the bottom of the CSV file that should not be used for inferring the schema. We should therefore remove any rows that might be part of the footer, which could happen if the CSV is small enough that the footers start less than 10 rows from the top. But we probably also want to make sure that we get 10 rows to infer the schema from, if the size of the CSV allows for that. We achieve that if we read `10+skipfooter` rows, and then remove the bottom `skipfooter` number of rows afterwards. It means a marginal efficiency loss, but should be negligible if `skipfooter` is not ridiculously high.